Look like the spare check failure is not due to this PR.
```
  CHECK   sound/soc/sof/intel/atom.c
sound/soc/sof/intel/atom.c:97:1: error: non-constant initializer for static object
sound/soc/sof/intel/atom.c:134:1: error: non-constant initializer for static object
sound/soc/sof/intel/atom.c:179:1: error: non-constant initializer for static object
sound/soc/sof/intel/atom.c:194:1: error: non-constant initializer for static object
sound/soc/sof/intel/atom.c:200:1: error: non-constant initializer for static object
sound/soc/sof/intel/atom.c:206:1: error: non-constant initializer for static object
sound/soc/sof/intel/atom.c:251:1: error: non-constant initializer for static object
sound/soc/sof/intel/atom.c:270:1: error: non-constant initializer for static object
sound/soc/sof/intel/atom.c:333:1: error: non-constant initializer for static object
sound/soc/sof/intel/atom.c:404:1: error: non-constant initializer for static object
sound/soc/sof/intel/atom.c:418:1: error: non-constant initializer for static object
make[6]: *** [scripts/Makefile.build:245: sound/soc/sof/intel/atom.o] Error 1
```

--- a/sound/soc/intel/boards/sof_rt5682.c
+++ b/sound/soc/intel/boards/sof_rt5682.c
@@ -340,8 +340,19 @@ static int sof_rt5682_hw_params(struct snd_pcm_substream *substream,
 		clk_id = RT5682_SCLK_S_PLL1;
 		break;
 	case CODEC_RT5682S:
-		pll_id = RT5682S_PLL2;
-		clk_id = RT5682S_SCLK_S_PLL2;
+		/*
+		 * For MCLK = 24.576MHz and sample rate = 96KHz case, use PLL1  We don't test
+		 * pll_out or params_rate() here since rt5682s PLL2 doesn't support 24.576MHz
+		 * input, so we have no choice but to use PLL1. Besides, we will not use PLL at
+		 * all if pll_in == pll_out. ex, MCLK = 24.576Mhz and sample rate = 48KHz
+		 */
+		if (pll_in == 24576000) {
+			pll_id = RT5682S_PLL1;
+			clk_id = RT5682S_SCLK_S_PLL1;
+		} else {
+			pll_id = RT5682S_PLL2;
+			clk_id = RT5682S_SCLK_S_PLL2;
+		}
 		break;
 	default:
 		dev_err(rtd->dev, "invalid codec type %d\n", ctx->codec_type);

